pkg/operator/controller/dns/controller.go

```diff
@@ -114,6 +114,7 @@ type Config struct {
 	// PrivateHostedZoneAWSEnabled indicates whether the "SharedVPC" feature gate is
 	// enabled.
 	PrivateHostedZoneAWSEnabled bool
+	CustomEndpointsEnabled      bool
```

Comment:
The new field should have "GCP" in the name in order to make it obvious that this is GCP-specific, and to avoid a potential conflict if we eventually add a similar featuregate for Azure (the other platforms already have APIs for spedcifying custom service endpoints, so it looks like Azure is the only potential future conflict).  The field should also have godoc.  
```suggestion
	// CustomEndpointsEnabled indicates whether the "GCPCustomAPIEndpoints"
	// featuregate is enabled.
	CustomEndpointsEnabled bool
```

Reply:
Thanks @Miciah I made those changes. One of the tests I believe is still failing so I need to look into that though.